### PR TITLE
Fix/rest api issues

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/81691

Some fixes needed to make sure the REST API code never throws any notice.

### CODE REVIEW

File: src/Tribe/REST/V1/Post_Repository.php [hunk 1]
```diff
@@ -511,7 +543,7 @@ public function prepare_term_data( $term_data, $taxonomy, $namespace ) {
 
 		unset( $term_data['_links'] );
 
-		$term_id = Tribe__Utils__Array::get( $term_data, 'id', $term_data['term_id'] );
+		$term_id = isset( $term_data['id'] ) ? $term_data['id'] : $term_data['term_id'];
```
Comment (Camwyn) [inline src/Tribe/REST/V1/Post_Repository.php]: After the `foreach ( $rename_map as $old => $new )` above, this _seems_ redundant?

Comment (lucatume) [inline src/Tribe/REST/V1/Post_Repository.php]: right!

File: src/Tribe/REST/V1/Post_Repository.php [hunk 2]
```diff
@@ -401,18 +410,32 @@ protected function get_categories( $event_id ) {
...
 		$terms_data = $this->prepare_terms_data( $terms, $taxonomy );
 
 		$data = array();
 		foreach ( $terms_data as $term_data ) {
-			$term_id = Tribe__Utils__Array::get( $term_data, 'id', $term_data['term_id'] );
+			if ( empty( $term_data['id'] ) && empty( $term_data['term_id'] ) ) {
+				continue;
+			}
+
+			$term_id = isset( $term_data['id'] ) ? $term_data['id'] : $term_data['term_id'];
```
Comment (Camwyn) [inline src/Tribe/REST/V1/Post_Repository.php]: Not sure why we don't do this earlier - in `prepare_terms_data` but 🤷‍♂️

**Comment by Camwyn** [comment]
🐫 💨

**Pull request merged**